equinix/resource_fabric_connection.go

```diff
@@ -0,0 +1,265 @@
+	ctx = context.WithValue(ctx, v4.ContextAccessToken, meta.(*Config).FabricAuthToken)
+	conn, _, err := client.ConnectionsApi.GetConnectionByUuid(ctx, d.Id(), nil)
+
+	if err != nil {
+		log.Printf("[WARN] Connection %s not found ", d.Id())
+		d.SetId("")
+		return diag.FromErr(err)
+	}
+	d.SetId(conn.Uuid)
+	return setFabricMap(d, conn)
+}
+
+func setFabricMap(d *schema.ResourceData, conn v4.Connection) diag.Diagnostics {
+	diags := diag.Diagnostics{}
+	err := setMap(d, map[string]interface{}{
+		"name":            conn.Name,
+		"bandwidth":       conn.Bandwidth,
+		"href":            conn.Href,
+		"description":     conn.Description,
+		"is_remote":       conn.IsRemote,
+		"type":            conn.Type_,
+		"state":           conn.State,
+		"direction":       conn.Direction,
+		"operation":       operationToTerra(conn.Operation),
+		"order":           orderMappingToTerra(conn.Order),
+		"change_log":      changeLogToTerra(conn.ChangeLog),
+		"redundancy":      redundancyToTerra(conn.Redundancy),
+		"notifications":   notificationToTerra(conn.Notifications),
+		"account":         accountToTerra(conn.Account),
+		"a_side":          connectionSideToTerra(conn.ASide),
+		"z_side":          connectionSideToTerra(conn.ZSide),
+		"additional_info": additionalInfoToTerra(conn.AdditionalInfo),
+		"project":         projectToTerra(conn.Project),
+	})
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	return diags
+}
+
+func resourceFabricConnectionUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	log.Printf(" inside fabric connection update resource ")
+	client := meta.(*Config).fabricClient
+	ctx = context.WithValue(ctx, v4.ContextAccessToken, meta.(*Config).FabricAuthToken)
+	uuid := d.Id()
+	if uuid == "" {
+		return diag.Errorf("No connection found for the value uuid %v ", uuid)
+	}
+	dbConn := v4.Connection{}
+	var err error
+	dbConn, err = waitUntilConnectionIsActive(uuid, meta, ctx)
+	if err != nil {
+		return diag.Errorf(" Either timed out or errored out while fetching connection for uuid %s and error %v", uuid, err)
+	}
+	update, err := getUpdateRequest(dbConn, d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	updates := []v4.ConnectionChangeOperation{update}
+	_, res, err := client.ConnectionsApi.UpdateConnectionByUuid(ctx, updates, uuid)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf(" Error response for the connection update, response %v, error %v", res, err))
+	}
+	updatedConn := v4.Connection{}
+	updatedConn, err = waitForConnectionUpdateCompletion(uuid, meta, ctx)
+
+	if err != nil {
+		return diag.FromErr(fmt.Errorf(" Errored while waiting for successful connection update, response %v, error %v", res, err))
+	}
+
+	d.SetId(updatedConn.Uuid)
+	return setFabricMap(d, updatedConn)
+}
+
+func waitForConnectionUpdateCompletion(uuid string, meta interface{}, ctx context.Context) (v4.Connection, error) {
+	log.Println("Waiting for connection update to complete")
```

Comment:
the uuid may be helpful in this log message

Reply:
Yes, added.